lib/phoenix/transports/long_poll.ex

```diff
@@ -261,18 +230,21 @@ defmodule Phoenix.Transports.LongPoll do
     Phoenix.Token.verify(endpoint, Atom.to_string(endpoint.__pubsub_server__), signed, opts[:crypto])
   end
 
-  defp status_json(conn, data) do
-    status = Plug.Conn.Status.code(conn.status || 200)
-    data   = Map.put(data, :status, status)
-    conn
-    |> put_status(200)
-    |> Phoenix.Controller.json(data)
+  defp status_json(conn) do
+    status = conn.status || 200
+    send_json(conn, "{\"status\":#{status}}")
   end
 
-  defp code_reload(conn, opts, endpoint) do
-    reload? = Keyword.get(opts, :code_reloader, endpoint.config(:code_reloader))
-    if reload?, do: Phoenix.CodeReloader.reload!(endpoint)
+  defp status_token_messages_json(conn, token, messages) do
+    status = conn.status || 200
+    messages = [?[, Enum.intersperse(messages, ?,), ?]]
+    token = Phoenix.json_library().encode_to_iodata!(token)
+    send_json(conn, "{\"status\":#{status},\"token\":#{token},\"messages\":#{messages}}")
```

Comment:
Is there any reason not to use a map here and `Phoenix.json_library().encode` the whole thing?

Reply:
If we do that, we need a special formatter only for long polling. I want to share them, which means they have to return already encoded data.